### PR TITLE
test(*) include plugin fixture Lua path

### PR DESCRIPTION
**Pull request by Tieske**
The custom plugins used for testing were invoked in different ways.
This change includes the path to the plugins in both the test engine (busted)
and in the test Kong instance. This way the plugins can simply
be inserted through the faster, and more reliable, dao-helpers instead
of having to do it through an http request.

I explicitly only included the custom plugins in the path and not the other
Lua files in the `fixtures` to prevent unexpected collisions.

### CODE REVIEW

File: spec/helpers.lua [hunk 1]
```diff
@@ -763,6 +768,22 @@ local function kong_exec(cmd, env)
...
   env = env or {}
 
+  -- Insert the Lua path to the custom-plugin fixtures
+  local lp
+  for k,v in pairs(env) do
+    if k:upper() == "LUA_PACKAGE_PATH" then
+      lp = k
+      break
+    end
+  end
```
Comment (thibaultcha) [inline spec/helpers.lua]: I think we can get rid of such a loop and the unnecessary cognitive load it requires and simply perform a lookup with both lower-case and upper-case keys. Upper-case is not supported anyways, so lower-case is even good enough. Having the `CUSTOM_PLUGIN_PATH` first would also be preferable.

```lua
if not env.lua_package_path then
  env.lua_package_path = CUSTOM_PLUGIN_PATH

else
  env.lua_package_path = CUSTOM_PLUGIN_PATH .. ";" .. env.lua_package_path  
end
```

File: spec/helpers.lua [hunk 2]
```diff
@@ -21,6 +22,10 @@ local log = require "kong.cmd.utils.log"
 
 log.set_lvl(log.levels.quiet) -- disable stdout logs in tests
 
+-- Add to package path so dao helpers can insert custom plugins
+-- (while running from the busted environment)
+package.path = package.path .. ";" .. CUSTOM_PLUGIN_PATH
```
Comment (thibaultcha) [inline spec/helpers.lua]: Better to have the custom path first.

File: spec/helpers.lua [hunk 3]
```diff
@@ -763,6 +768,22 @@ local function kong_exec(cmd, env)
...
+  for k,v in pairs(env) do
+    if k:upper() == "LUA_PACKAGE_PATH" then
+      lp = k
+      break
+    end
+  end
+  if not lp then
+    lp = "lua_package_path"
+    env[lp] = CUSTOM_PLUGIN_PATH
+  else
```
Comment (thibaultcha) [inline spec/helpers.lua]: style: missing line jump

File: spec/helpers.lua [hunk 4]
```diff
@@ -763,6 +768,22 @@ local function kong_exec(cmd, env)
...
+    env[lp] = env[lp] ..";" .. CUSTOM_PLUGIN_PATH
```
Comment (thibaultcha) [inline spec/helpers.lua]: style: missing space

Comment (Tieske) [inline spec/helpers.lua]: @thibaultcha comments fixed!

**Pull request merged**